_posts/2024-01-23-dynamic-github-branch-protection-status-checks.markdown

```diff
@@ -0,0 +1,263 @@
+---
+layout: single
+title: "Dynamic GitHub Branch Protection Status Checks"
+tags: technology github devops
+date: 2024-01-23 00:00:00 +0100
+author: David Werner  # as used in `authors.yml`
+author_profile: true
+header:
+  overlay_image: assets/images/tech_gear_banner.jpg  # can also be a different asset
+  overlay_filter: 0.2
+  show_overlay_excerpt: false
+---
+
+# Dynamic GitHub Branch Protection Status Checks
+
```

Comment:
Probably a very high level introduction would be good, something like:
In CI/CD you want all changes be automatically tested. Sometimes, however some tests might be very costly or take very long, in this case you might want to execute those tests only if neccessary, that is, if certain conditions are met.
In this post we will show you how to achieve this in Github Actions.

Reply:
Good point! And I was planning to add one anyway. 
That's why it's only a draft pull request, yet. 😉